### PR TITLE
Move a Project to an Organization

### PR DESCRIPTION
**Pull request by inpaceee**

### CODE REVIEW

File: packages/backend/src/graphql/organization/organization.ts [hunk 1]
```diff
@@ -3,14 +3,16 @@ import { ModifyInterface } from '../interfaces/modifyInterface'
...
+  select: { id: true },
   interfaces: [ModifyInterface],
   fields: (t) => ({
     id: t.exposeID('id', { description: 'identifies the organization' }),
     title: t.exposeString('title'),
     address: t.exposeString('address', {
+      authScopes: (organization) => ({ isAdminByOrganization: organization.id }),
       nullable: true,
     }),
+    projects: t.relation('projects'),
```
Comment (PascalHelbig) [inline packages/backend/src/graphql/organization/organization.ts:15]: add auth scope

File: packages/backend/src/graphql/builder.ts [hunk 2]
```diff
@@ -77,43 +78,45 @@ export const builder = new SchemaBuilder<{
...
         where: {
-          userId: context.session.user.id,
-          project: { tasks: { some: { id: taskId } } },
+          tasks: { some: { id: taskId } },
+          ...getWhereUserIsMember(context.session.user.id),
         },
       })
 
-      return !!projectMembership
+      return !!project
```
Comment (PascalHelbig) [inline packages/backend/src/graphql/builder.ts:119]: Maybe for consistency, use `prisma.task.findFirst`

File: packages/backend/src/graphql/builder.ts [hunk 3]
```diff
@@ -167,17 +163,14 @@ export const builder = new SchemaBuilder<{
         return false
       }
 
-      const projectMembership = await prisma.projectMembership.findFirst({
-        select: { role: true },
+      const project = await prisma.project.findFirst({
```
Comment (PascalHelbig) [inline packages/backend/src/graphql/builder.ts:166]: see comment above

File: packages/backend/src/graphql/project/mutations/projectUpdateMutationField.ts [hunk 4]
```diff
@@ -11,14 +11,15 @@ builder.mutationField('projectUpdate', (t) =>
       data: t.arg({ type: ProjectInput }),
     },
     authScopes: async (_source, { id }) => ({ isAdminByProject: id.toString() }),
```
Comment (PascalHelbig) [inline packages/backend/src/graphql/project/mutations/projectUpdateMutationField.ts:13]: When updating the organization, we should check that the user is org member of the organization.

File: packages/backend/src/graphql/project/mutations/projectCreateMutationField.ts [hunk 5]
```diff
@@ -9,13 +9,14 @@ builder.mutationField('projectCreate', (t) =>
     args: {
       data: t.arg({ type: ProjectInput }),
     },
```
Comment (PascalHelbig) [inline packages/backend/src/graphql/project/mutations/projectCreateMutationField.ts:11]: add authScope that checks if the user is org member of the given organization.

File: apps/web/frontend/components/projectForm/projectForm.tsx [hunk 6]
```diff
@@ -209,6 +229,31 @@ export const ProjectForm = (props: ProjectFormProps): JSX.Element => {
...
+                Select an organization
+              </option>
+              {organizationsData?.organizations.map((organization) => (
+                <option key={organization.id} value={organization.id}>
+                  {organization.title}
+                </option>
+              ))}
+              {project?.organization?.isArchived && (
+                <option value={project.organization.id} disabled>
+                  {project.organization.title} (archived)
```
Comment (PascalHelbig) [inline apps/web/frontend/components/projectForm/projectForm.tsx:251]: show archived label only when it is really archived

File: apps/web/frontend/components/projectForm/projectForm.tsx [hunk 7]
```diff
@@ -67,10 +68,24 @@ export const ProjectFormFragment = graphql(`
...
 `)
 
+export const OrganizationsQueryDocument = graphql(`
+  query organizations {
+    organizations {
+      id
+      title
+    }
+  }
+`)
```
Comment (PascalHelbig) [inline apps/web/frontend/components/projectForm/projectForm.tsx:87]: Use fragment instead and do the query outside the `ProjectForm` component

File: apps/web/frontend/components/projectForm/projectForm.test.tsx [hunk 8]
```diff
@@ -142,10 +152,47 @@ describe('projectForm', () => {
...
+
+    const organizationSelect = screen.getByRole('combobox', { name: /organization/i })
+    expect(organizationSelect).toBeInTheDocument()
+    // userEvent.selectOptions(organizationSelect, '')
+    // expect(onSubmit).toHaveBeenNthCalledWith(1, {
+    //   end: null,
+    //   start: '2022-04-12',
+    //   title: 'test project',
+    //   organizationId: '',
+    // })
```
Comment (PascalHelbig) [inline apps/web/frontend/components/projectForm/projectForm.test.tsx:196]: don't forget

**Review by YohannesTesfay — APPROVED** [review]
shaun-the-sheep-movie-ok-gif

**Pull request merged**